Logger/LogFactory.cs

```diff
@@ -2,10 +2,27 @@
 {
     public class LogFactory
     {
+        private string? path;
         public BaseLogger CreateLogger(string className)
```

Comment:
```suggestion
        public BaseLogger? CreateLogger(string className)
```
Declare this as nullable, that way you don't use the `!` operator all over the place.